### PR TITLE
New package: gek-0.1

### PR DESCRIPTION
**Pull request by JohnGebbie**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packagegs/blob/master/Manual.md#quality-requirements): **YES**

### CODE REVIEW

File: srcpkgs/gek/template [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+version=0.1
+revision=1
+hostmakedepends="git go"
+short_desc="Stack-based stream editor and scripting language"
+maintainer="John Gebbie <me@johngebbie.com>"
+license="GPL-3.0-only"
+homepage="https://geklang.org/"
+
+do_fetch() {
+	git clone https://git.sr.ht/~geb/gek "${wrksrc}"
```
Comment (classabbyamp) [inline srcpkgs/gek/template:12]: git packages are not accepted. the software must have a release

File: srcpkgs/gek/template [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	cd "${wrksrc}"
+}
+
+do_build() {
+	go build -o gek ./src
```
Comment (classabbyamp) [inline srcpkgs/gek/template:17]: you'll want to use the go build_style, probably

File: srcpkgs/gek/template (hunk 1 above)
Comment (JohnGebbie) [reply srcpkgs/gek/template]: I've got it using --branch v0.1 now, I hope that's good.

Comment (classabbyamp) [reply srcpkgs/gek/template]: you should be using `distfiles`, not `do_fetch()`, i.e.: `distfiles="https://git.sr.ht/~geb/gek/archive/v${version}.tar.gz"`

Comment (JohnGebbie) [reply srcpkgs/gek/template]: bum. I'm not sure how to do that. Do I put archives in the repo manually?  I copied the --branch approach having greped the other templates.

Comment (classabbyamp) [reply srcpkgs/gek/template]: sr.ht should auto-generate a tarball when you create a tag (https://git.sr.ht/~geb/gek/archive/v0.1.tar.gz should work)

those `git clone`s in other templates are either very old or special cases

Comment (paper42) [reply srcpkgs/gek/template]: Delete do_fetch and let xbps-src use the default one. The default do_fetch function downloads the archive in $distfiles. Look at almost every single template, almost no other template uses git. For example the `vale` package template will have a similar structure.

Comment (JohnGebbie) [reply srcpkgs/gek/template]: Thank you. I think I've done it.

File: srcpkgs/gek/template [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+maintainer="John Gebbie <me@johngebbie.com>"
+license="GPL-3.0-only"
+homepage="https://geklang.org/"
+distfiles="https://git.sr.ht/~geb/gek/archive/v${version}.tar.gz"
+checksum="3a271b067793099a9ad873cd6f9113b3bdc376160e1bdb9f72892e94d0c8f1c6"
+
+# The default would install it called src.
+do_build() {
+	go build -o gek "${go_package}"
+}
```
Comment (paper42) [inline srcpkgs/gek/template:19]: you can easily rename it in do/post_install

Comment (JohnGebbie) [reply srcpkgs/gek/template]: Ok I've done it. I'm going to bed now, thanks for the heads ups.

File: srcpkgs/gek/template [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+homepage="https://geklang.org/"
+distfiles="https://git.sr.ht/~geb/gek/archive/v${version}.tar.gz"
+checksum="3a271b067793099a9ad873cd6f9113b3bdc376160e1bdb9f72892e94d0c8f1c6"
+
+do_install() {
+	vbin "$GOPATH/bin/src" gek
+	vbin gekdoc
+	vman gek.1
+	mkdir "${DESTDIR}/usr/share/doc"
+	cp -r doc "${DESTDIR}/usr/share/doc/${pkgname}"
```
Comment (paper42) [inline srcpkgs/gek/template:21]:
```suggestion
	vmkdir /usr/share/doc
	vcopy doc "/usr/share/doc/${pkgname}"
```

**Comment by classabbyamp** [comment]
```
=> ERROR: gek-0.1_1: vinstall: cannot find '/builddir/gek-v0.1/_build-gek-xbps/bin/src'...
```

I believe go puts things in a different dir for cross builds.

this is what the build style does:

https://github.com/void-linux/void-packages/blob/a12a5fa7a4ad36fc18dc1254c6c9ffc806777da3/common/build-style/go.sh#L43-L49

File: srcpkgs/gek/template [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+homepage="https://geklang.org/"
+distfiles="https://git.sr.ht/~geb/gek/archive/v${version}.tar.gz"
+checksum="3a271b067793099a9ad873cd6f9113b3bdc376160e1bdb9f72892e94d0c8f1c6"
+
+do_install() {
+	if [ -f "$GOPATH/bin/src" ]; then
+		vbin "$GOPATH/bin/src" gek
+	else
+		vbin "$GOPATH/bin/**/src" gek
+	fi
```
Comment (paper42) [inline srcpkgs/gek/template:21]:
```suggestion
post_install() {
	mv ${DESTDIR}/usr/bin/src ${DESTDIR}/usr/bin/gek
```

**Comment by JohnGebbie** [comment]
I'll close this, my program could do with some work, and I don't want to add to the backlog. Thanks for helping me learn the xbps-src.

**Pull request closed without merging**